src/lndk_offers.rs

```diff
@@ -528,8 +588,31 @@ impl InvoicePayer for Client {
             blinded_hops.push(new_hop);
         }
 
+        let introduction_node = match path.introduction_node {
+            IntroductionNode::NodeId(pubkey) => pubkey,
+            IntroductionNode::DirectedShortChannelId(direction, scid) => {
+                let get_info_request = ChanInfoRequest { chan_id: scid };
```

Comment:
Looks like this can also be covered by the `get_chan_info` function you created, if you can change it to not be a method